core/src/main/java/hudson/model/User.java

```diff
@@ -713,8 +714,9 @@ public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
         if (id == null || StringUtils.isBlank(id)) {
             return false;
         }
+        final String trimmedId = id.trim();
         for (String invalidId : ILLEGAL_PERSISTED_USERNAMES) {
-            if (id.equalsIgnoreCase(invalidId))
+            if (trimmedId.equalsIgnoreCase(invalidId))
```

Comment:
\u041d is not a whitespace character nor is it trailing... it is the "Cryllic Capital Letter En" (Н) [wikipedia](https://en.wikipedia.org/wiki/En_(Cyrillic)).

It would fail the trimmedId.equalsIgnoreCase(invalidId) check and as such be classed as a valid username.